helm/lint.go

```diff
@@ -22,80 +30,198 @@ type KubeObject struct {
+			"(in this case, '%v'). Use .Release.Name in your template to ensure that all objects are named with the release "+
+			"as a suffix to aovid name collisions across releases.",
+			kind, meta.Name, suffix)
 		errors = append(errors, e)
 	}
-	ctx.Logger.Debugf("Object with kind '%v' and name '%v': object name does indeed contain the desired suffix `%v`", obj.Kind, obj.Metadata.Name, suffix)
+	ctx.Logger.Debugf("Object with kind '%v' and name '%v': object name does indeed contain the desired suffix `%v`",
+		kind, meta.Name, suffix)
 
 	// Verify that every object is labeled with a key `release` and value equal to the current context's release
-	if obj.Metadata.Labels["release"] != release {
-		e := fmt.Errorf("Object with kind '%v' and name '%v': object is missing a `release` label with the release name as a value (in this case, '%v'). Found these labels on the object: %+v", obj.Kind, obj.Metadata.Name, release, obj.Metadata.Labels)
+	if meta.Labels["release"] != release {
+		e := fmt.Errorf("Object with kind '%v' and name '%v': object is missing a `release` label with the "+
+			"release name as a value (in this case, '%v'). Found these labels on the object: %+v",
+			kind, meta.Name, release, meta.Labels)
 		errors = append(errors, e)
 	}
-	ctx.Logger.Debugf("Object with kind '%v' and name '%v': object labels exist, and the release label is '%v'", obj.Kind, obj.Metadata.Name, obj.Metadata.Labels["release"])
+	ctx.Logger.Debugf("Object with kind '%v' and name '%v': object labels exist, and the release label is '%v'",
+		kind, meta.Name, meta.Labels)
+
+	return errors
+}
+
+func LintObject(ctx *ankh.ExecutionContext, obj runtime.Object) []error {
+	errors := []error{}
+	release := ctx.AnkhConfig.CurrentContext.Release
+
+	switch obj.(type) {
+
+	case *appsv1beta1.Deployment:
+		deployment := obj.(*appsv1beta1.Deployment)
+
+		metaErrors := LintObjectMeta(ctx, deployment.ObjectMeta, deployment.Kind)
+		if len(metaErrors) > 0 {
+			errors = append(errors, metaErrors...)
+		}
 
-	switch strings.ToLower(obj.Kind) {
-	case "deployment":
 		// The Deployment should create pods with the `release` label
-		if obj.Spec.Template.Metadata.Labels["release"] != release {
-			e := fmt.Errorf("Deployment with name '%v': object's spec.template.metadata.labels is missing a `release` label with the release name as a value (in this case, '%v'). Found these labels on spec.template.metadata: %+v", obj.Kind, obj.Metadata.Name, release, obj.Spec.Template.Metadata.Labels)
+		if deployment.Spec.Template.ObjectMeta.Labels["release"] != release {
+			e := fmt.Errorf("Deployment with name '%v': object's spec.template.metadata.labels is missing a `release` "+
+				"label with the release name as a value (in this case, '%v'). Found these labels on spec.template.metadata: %+v",
+				deployment.Kind, deployment.ObjectMeta.Name, release, deployment.Spec.Template.ObjectMeta.Labels)
```

Comment:
Looks like too many args to `Errorf` here

Reply:
Oops, thanks!